test/java/com/odysseusinc/arachne/portal/front/StudyManagerTest.java

```diff
@@ -1,61 +1,699 @@
+        editButton.click();
 
-        final By modal = ByBuilder.modal(MODAL_TITLE_CREATE_STUDY);
-        waitFor(driver, modal);
+        WebElement description = driver.findElement(ByBuilder.textArea("Add Study Objective"));
+        Assert.assertEquals("", description.getText());
 
-        WebElement modalWebElement = driver.findElement(modal);
-        WebElement studyNameInput = modalWebElement.findElement(ByBuilder.input(PLACEHOLDER_STUDY_TITLE));
-        WebElement studyTypeSelect = modalWebElement.findElement(ByBuilder.select(PLACEHOLDER_STUDY_TYPE));
-        WebElement studyTypeOption = modalWebElement.findElement(ByBuilder.selectOption(studyType));
-        WebElement studyCreateBtn = modalWebElement.findElement(ByBuilder.button("Create"));
+        String newDescription = "Long and beautiful description";
+        description.sendKeys(newDescription);
 
-        studyNameInput.sendKeys(studyName);
+        WebElement studyDescriptionBtn = driver
+                .findElement(By.className("ac-study-objective-edit"))
+                .findElement(ByBuilder.byClassAndText("ac-form__submit", "Save"));
+
+        studyDescriptionBtn.click();
+
+        waitFor(driver, ByBuilder.byClassAndText("ac-study-objective-view", newDescription));
+    }
+
+    @Test
+    public void test04UpdateStudyType() throws Exception {
+
+        loginAndOpenStudy();
+
+        String studyType = "Clinical Trial Design";
+
+        WebElement studyTypeSelect = driver.findElement(ByBuilder.select(studyType));
+        waitFor(driver, ByBuilder.select(studyType));
+
+        String updatedStudyType = "Health Economics and Outcomes";
+        WebElement studyTypeOption = driver.findElement(ByBuilder.selectOption(updatedStudyType));
 
         Actions actions = new Actions(driver);
         actions.moveToElement(studyTypeSelect).click().build().perform();
         actions.moveToElement(studyTypeOption).click().build().perform();
 
-        studyCreateBtn.click();
+        waitFor(driver, ByBuilder.select(updatedStudyType));
+    }
+
+    @Test
+    public void test05AddStudyDocument() throws Exception {
+
+        loginAndOpenStudy();
+
+        WebElement studyTypeSelect = driver.findElement(ByBuilder.byClassAndText("ac-study-document-list-add__label", "Add document"));
+        studyTypeSelect.click();
+        waitFor(driver, ByBuilder.byClassAndText("ac-tabs__item ac-tabs__item--selected", "Computer"));
+
+        WebElement uploadElement = driver.findElement(By.cssSelector("input[type=file]"));
+        uploadElement.sendKeys(file.getAbsolutePath());
+        driver.findElement(ByBuilder.byClassAndText("ac-form__submit", "Add")).click();
+        waitFor(driver, ByBuilder.byClassAndText("ac-link ac-code-file-info__name", "text.txt"));
+    }
+
+    @Test
+    public void test06RemoveStudyDocument() throws Exception {
+
+        loginAndOpenStudy();
+        waitFor(driver, By.className("ac-list-item__remove"));
```

Comment:
This is too general search. Such remove class can appear in Analysis list, too. SSo find parent panel first and then search inside it